File: AD-HOC/1936/main.c
#include <stdio.h>

int smallestFactorialSum(int n) {
    int k = 0;
    int factorials[8] = {40320, 5040, 720, 120, 24, 6, 2, 1};
    for (int i = 0; i < 8; i++) {
        int a = factorials[i];
        while (n >= a) {
            n -= a;
            k += 1;
        }
        if (n == 0) break;
    }
    return k;
}

int main() {
    int n;
    scanf("%d", &n);
    printf("%d\n", smallestFactorialSum(n));
    return 0;
}

File: AD-HOC/1936/main.dart
import 'dart:io';

int smallestFactorialSum(int n) {
  int k = 0;
  List<int> factorials = [40320, 5040, 720, 120, 24, 6, 2, 1];
  for (int a in factorials) {
    while (n >= a) {
      n -= a;
      k += 1;
    }
    if (n == 0) break;
  }
  return k;
}

void main() {
  int n = int.parse(stdin.readLineSync()!);
  print(smallestFactorialSum(n));
}
